webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/migration/MigrateNodeAcceptationInventories.scala

```diff
@@ -0,0 +1,166 @@
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package bootstrap.liftweb.checks.migration
+
+import bootstrap.liftweb.BootstrapChecks
+import bootstrap.liftweb.BootstrapLogger
+import com.normation.errors.IOResult
+import com.normation.inventory.domain.FullInventory
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.db.Doobie
+import com.normation.rudder.domain.logger.MigrationLoggerPure
+import com.normation.rudder.facts.nodes.NodeFact
+import com.normation.rudder.services.nodes.NodeInfoService
+import com.normation.rudder.services.nodes.history.HistoryLogRepository
+import com.normation.rudder.services.nodes.history.impl.FactLog
+import com.normation.rudder.services.nodes.history.impl.InventoryHistoryLogRepository
+import com.normation.zio._
+import doobie.implicits._
+import org.apache.commons.io.FileUtils
+import org.joda.time.DateTime
+import zio._
+import zio.interop.catz._
+
+/*
+ * Before Rudder 8.0, we used to save the state of node when accepted in an LDIF file under
+ *   /var/rudder/inventories/historical/${nodeid}/${iso-date-time-of-acceptation}
+ * Since 8.0, we store them in the nodefacts tables, which is create in the process.
+ * Ths migration is convergent and asynchronous:
+ * - it does not block boot
+ * - it can be interrupted and restarted afterward
+ * During migration, user won't be able to access the history inventory of the node.
+ *
+ * The migration is also a good time to clean-up that directory, and actually an empty directory will signal
+ * end of migration:
+ * - for each directory under `/var/node/historical/`,
+ * - if the node still exist OR if the event is less than KEEP_REFUSED_DURATION, add it in the base else nothing
+ * - delete the directory
+ */
+class MigrateNodeAcceptationInventories(
+    nodeInfoService:   NodeInfoService,
+    doobie:            Doobie,
+    fileLogRepository: InventoryHistoryLogRepository,
+    jdbcLogRepository: HistoryLogRepository[NodeId, DateTime, NodeFact, FactLog],
+    MAX_KEEP_REFUSED:  Duration
+) extends BootstrapChecks {
+
+  import doobie._
+
+  val msg = "old inventory accept/refuse facts to 'NodeFacts' database table"
+
+  override def description: String =
+    "Check if table 'NodeFacts' exists and if data from /var/rudder/inventories/historical are migrated"
+
+  def createTableStatement: IOResult[Unit] = {
+    val sql = sql"""CREATE TABLE IF NOT EXISTS NodeFacts (
+        nodeId           text PRIMARY KEY
+      , acceptRefuseDate timestamp with time zone
```

Comment:
Why not simply date ? and fact ?

Reply:
because I want to be able to have latter on `date` and `fact` for the current state of a node, and so to be able to clearly differentiate them appart (actually, it will likely be `modifyDate` and `currentFact` or something like that)